src/GameSync.Models/Users/User.cs

```diff
@@ -0,0 +1,36 @@
+﻿using GameSync.Models.Users.ValueObjects;
+
+namespace GameSync.Models.Users;
+public class User
+{
+    public UserId UserId { get; private set; }
+    public UserName UserName { get; private set; }
+    public Password Password { get; private set; }
+
+    public User(
+        UserId userId,
+        UserName userName,
+        Password password
+    )
+    {
+        SetUserIdOrThrowException(userId);
```

Comment:
Invokation in constructor suggests logic applied. How to replace it with following snippet?
`UserId = userId ?? throw new ArgumentNullException(nameof(userId));`

Reply:
You are right. I was a bit too much 'for the future'

Reply:
Fixed